analysis/quantifier-comparisons/13-spatial-transcriptomics-benchmarking.Rmd

```diff
@@ -0,0 +1,411 @@
+  
+  # arrange plots and add sample name as title 
+  grid.arrange(p1, p2, nrow = 1, top = grid::textGrob(sample))
+}
+
+
+plot_spaceranger_spe <- function(spe, sample, column){
+  # plot spots only 
+  p1 <- ggspavis::plotSpots(spe, 
+                            # switch the order of row and col to flip the image 
+                            x_coord = "pxl_row_in_fullres", 
+                            y_coord = "pxl_col_in_fullres", 
+                            annotate = column) +
+    scale_color_viridis_c() 
+  
+  # plot with tissue underneath
+  p2 <- ggspavis::plotVisium(spe, 
+                                   x_coord = "pxl_row_in_fullres", 
+                                   y_coord = "pxl_col_in_fullres", 
+                                   fill = column) +
+    scale_fill_viridis_c()
+  
+  # arrange plots and add sample name as title 
+  grid.arrange(p1, p2, nrow = 1, top = grid::textGrob(sample))
+  
+}
+```
+
+First we will look at the per cell metrics: mitochondrial reads per cell, total UMI per cell, and total genes detected per cell. 
+
+```{r}
+# % mitochondrial reads/ spot 
+ggplot(coldata_df_common, aes(x = tool, y = subsets_mito_percent, fill = tool)) + 
+  geom_boxplot() +
+  theme_classic() + 
+  theme(axis.text.x = element_text(angle = 45, hjust = 1)) + 
+  ylab("Mito Percent") + 
+  xlab("")
+```
+```{r}
+plot_fry_spe(all_spe_filter$`alevin-fry`, "alevin-fry", "subsets_mito_percent")
+```
+```{r}
+plot_spaceranger_spe(all_spe_filter$spaceranger, "spaceranger", "subsets_mito_percent")
+```
+Overall it looks like mitochondrial content is low and fairly similar across both tools. 
+
+
+```{r}
+# total UMI/ spot 
+ggplot(coldata_df_common, aes(x = sum, color = tool)) + 
+  geom_density() + 
+  theme_classic() + 
+  theme(axis.text.x = element_text(angle = 45, hjust = 1)) + 
+  ylab("UMI/spot") + 
+  xlab("")
+```
+```{r}
+plot_fry_spe(all_spe_filter$`alevin-fry`, "alevin-fry", "sum")
+```
+
+```{r}
+plot_spaceranger_spe(all_spe_filter$spaceranger, "spaceranger", "sum")
+```
+
+```{r}
+# total genes/ spot 
+ggplot(coldata_df_common, aes(x = detected, color = tool)) + 
+  geom_density() + 
+  theme_classic() + 
+  theme(axis.text.x = element_text(angle = 45, hjust = 1)) + 
+  ylab("Genes detected/spot") + 
+  xlab("") 
+```
+```{r}
+plot_fry_spe(all_spe_filter$`alevin-fry`, "alevin-fry", "detected")
+```
+
+```{r}
+plot_spaceranger_spe(all_spe_filter$spaceranger, "spaceranger", "detected")
+```
+Generally it looks like both tools are fairly similar in the number of UMI/cell and genes detected/cell.
+
+### Per Gene QC Metrics
+
+Let's also look at the correlation of mean gene expression across shared genes. 
+We will first need to calculate the per feature QC on the filtered spes after removing spots not present in the tissue and then grab the `rowData` and combine into a data frame used for plotting. 
+
+```{r}
+all_spe_filter <- all_spe_filter %>%
+  purrr::map(scuttle::addPerFeatureQCMetrics)
```

Comment:
Did these get removed when we filtered? And/or should we just skip adding them at the earlier step (Line 117)?